lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+            if is_complete('kiosk_registration_4c7f', redcap_record_instance):
+                collection_method = CollectionMethod.KIOSK
+            elif is_complete('test_fulfillment_form', redcap_record_instance):
+                collection_method = CollectionMethod.SWAB_AND_SEND
+
+        # site_reference refers to where the sample was collected
+        site_reference = create_site_reference(redcap_record_instance, collection_method, event_type)
+
+        initial_encounter_entry, initial_encounter_reference = create_encounter(
+            redcap_record_instance, patient_reference, site_reference,
+            location_resource_entries, event_type, collection_method)
+
+        # Skip the entire record if we can't create the enrollment encounter.
+        # Otherwise, just skip the record instance.
+        if not initial_encounter_entry:
+            if event_type == EventType.ENROLLMENT:
+                return None
+            else:
+                LOG.warning("Skipping record instance with insufficient information to construct a initial encounter")
+                continue
+
+        specimen_entry = None
+        specimen_observation_entry = None
+        specimen_received = (collection_method == CollectionMethod.SWAB_AND_SEND and \
+            is_complete('post_collection_data_entry_qc', redcap_record_instance)) or \
+            (collection_method == CollectionMethod.KIOSK and \
+            is_complete('kiosk_registration_4c7f', redcap_record_instance))
+
+        if specimen_received:
+            specimen_entry, specimen_reference = create_specimen(redcap_record_instance, patient_reference, collection_method)
+            specimen_observation_entry = create_specimen_observation_entry(
+                specimen_reference, patient_reference, initial_encounter_reference)
+        else:
+            LOG.info("Creating encounter for record instance without sample")
+
+        if specimen_received and not specimen_entry:
+            LOG.warning("Skipping record instance with insufficent information to construct a specimen")
+            continue
+
+        computed_questionnaire_entry = None
+        enrollment_questionnaire_entry = None
+        daily_questionnaire_entry = None
+        testing_determination_internal_questionnaire_entry = None
+
+        computed_questionnaire_entry = create_computed_questionnaire_response(
+            redcap_record_instance, patient_reference, initial_encounter_reference,
+            birthdate, datetime.strptime(initial_encounter_entry['resource']['period']['start'], '%Y-%m-%d'))
+
+        if event_type == EventType.ENROLLMENT:
+            enrollment_questionnaire_entry = create_enrollment_questionnaire_response(
+            enrollment, patient_reference, initial_encounter_reference)
+        else:
+            testing_determination_internal_questionnaire_entry = \
+                create_testing_determination_internal_questionnaire_response(
```

Comment:
I'm not sure I understand why this testing determination field is being handled differently and put into a separate QuestionnaireResponse Resource than the others.

Reply:
This questionnaire is the internal questionnaire that is needed for the priority queue/quota. We need it for every encounter.
I'm treating it differently from the daily survey because it's used for differently purposes. The two questionnaires are conceptually not the same.

Reply:
Thanks. Could you add a description to the doc string of this function to elucidate the motivation for it?

Reply:
FYI, on the [FHIR ETL side ](https://github.com/seattleflu/id3c/blob/master/lib/id3c/cli/command/etl/fhir.py#L452) all of the QuestionnaireResponse Resources get combined into one.

Reply:
Good point. This is a bizarre instrument. I added this to the doc string:
This instrument is used
    to communicate with the Priority Queue application. Note that for this
    questionnaire we do want to return answers for all questions even if they are
    not answered.